pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -368,11 +376,25 @@ func (i *scaleUpdatedObjectInfo) UpdatedObject(ctx context.Context, oldObj runti
 		return nil, errors.NewNotFound(apps.Resource("deployments/scale"), i.name)
 	}
 
+	managedFieldsHandler, err := fieldmanager.NewScaleHandler(
+		deployment.ManagedFields,
+		schema.GroupVersion{Group: "apps", Version: "v1"},
+		replicasPathInDeployment,
+	)
+	if err != nil {
+		return nil, err
+	}
+
 	// deployment -> old scale
```

Comment:
Instead of constructing a scale object from the existing deployment, I expected us to construct a deployment object from the new scale object and the existing deployment object, and apply or update that?

Reply:
This is more or less what's happening. The problem is that in order to get the new scale object we need the "old scale" object from the deployment and run it through [UpdatedObject](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/apps/deployment/storage/storage.go#L378). UpdatedObject will update the managed fields of the scale object and conflict if needed.

Reply:
I think I am objecting to a large part of the existing codebase here. Antoine attempted to convince me that I have to hold my nose for this. I'll look again.